torch/nn/functional.py

```diff
@@ -1051,10 +1056,13 @@ def softmax(input, dim=None, _stacklevel=3, dtype=None):
     """
     if dim is None:
         dim = _get_softmax_dim('softmax', input.dim(), _stacklevel)
+    dim = torch.jit._unwrap_optional(dim)
```

Comment:
What does this function do? Throws if it’s not supplied or what?

Reply:
In Python it asserts that the param is not `None` and returns it, and in script its added as a builtin to do the same thing + convert an optional to its contained type